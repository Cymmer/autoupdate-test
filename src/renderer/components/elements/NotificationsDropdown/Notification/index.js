import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import styles from './styles.module.scss';

const Notification = ({ children, className, viewAction }) => (
  <button
    className={cn(styles.Notification, className)}
    data-test="notificationButton"
    type="button"
    onClick={viewAction}
  >
    {children}
  </button>
);

Notification.defaultProps = {
  className: null,
};

Notification.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,

  // opens the modal to view the notification
  viewAction: PropTypes.func.isRequired,
};

export default Notification;
