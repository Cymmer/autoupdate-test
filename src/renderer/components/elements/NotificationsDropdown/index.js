import React, { useState, useRef } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import GLOBALS from 'codechum-app-globals';
import styles from './styles.module.scss';

import Card from '../Card';
import IconButton from '../Button/IconButton';
import iconButtonTypes from '../Button/constants/iconButtonTypes';
import Text from '../Text';
import textTypes from '../Text/constants/textTypes';

import { useOnClickOutside } from '../../../hooks';

const NotificationsDropdown = ({ children, className }) => {
  const ref = useRef();
  const [isDropdownToggled, toggleDropdown] = useState(false);
  useOnClickOutside(ref, () => toggleDropdown(false));

  return (
    <div ref={ref} className={cn(styles.NotificationsDropdown, className)}>
      <IconButton
        className={cn(styles.NotificationsDropdown_button, {
          [styles.NotificationsDropdown_button___hasNotifications]:
            children?.length > 0,
        })}
        icon="notifications"
        type={iconButtonTypes.SOLID.MD}
        onClick={() => toggleDropdown(!isDropdownToggled)}
      />
      <Card
        className={cn({
          [styles.NotificationsDropdown_dropdown]: !isDropdownToggled,
          [styles.NotificationsDropdown_dropdown___toggled]: isDropdownToggled,
        })}
      >
        <div className={styles.NotificationsDropdown_head}>
          <Text type={textTypes.DATA.SM}>Notifications</Text>
        </div>
        <div className={styles.NotificationsDropdown_body}>
          {children?.length > 0 ? (
            <>{children}</>
          ) : (
            <div className={styles.NotificationsDropdown_empty}>
              <Text
                colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['400']}
                type={textTypes.BODY.SM}
              >
                No Notifications
              </Text>
            </div>
          )}
        </div>
      </Card>
    </div>
  );
};

NotificationsDropdown.defaultProps = {
  className: null,
  children: null,
};

NotificationsDropdown.propTypes = {
  className: PropTypes.string,
  children: PropTypes.arrayOf(PropTypes.node),
};

export default NotificationsDropdown;
