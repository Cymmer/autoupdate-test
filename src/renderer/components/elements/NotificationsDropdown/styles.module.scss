@import '../../../scss/variables';
@import '../../../scss/mixins';
@import '../../../scss/themes';

@mixin dropdown($display) {
  display: $display;
  flex-direction: column;
  position: absolute;
  top: 52px;
  right: 0;
  box-shadow: $shadow-raised;
  width: 300px;
  border: none;
  overflow: hidden;

  @include breakpoint(sm) {
    width: 240px;
    right: -60px;
  }

  @content;
}

.NotificationsDropdown {
  position: relative;

  &_dropdown {
    @include dropdown(none);

    &___toggled {
      @include dropdown(flex);
    }
  }

  &_head {
    padding: 8px 20px 4px 20px;
    border-bottom: 1px solid;
    @include theme {
      border-color: t('borderColor');
    }
  }

  &_body {
    overflow: auto;
    max-height: 240px;
  }

  &_button {
    position: relative;

    &___hasNotifications {
      &:after {
        content: '';
        width: 10px;
        height: 10px;
        background-color: $red-300;
        border-radius: 100%;
        border: 2px solid;
        @include theme {
          border-color: t('cardBackgroundColor');
        }
        position: absolute;
        top: 4px;
        right: 2px;
      }
    }
  }

  &_empty {
    padding: 20px;
    text-align: center;
  }
}
