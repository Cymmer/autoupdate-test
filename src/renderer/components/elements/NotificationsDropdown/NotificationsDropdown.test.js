import React from 'react';
import { shallow } from 'enzyme';
import NotificationsDropdown from './index';

describe('RENDERING - NotificationsDropdown', () => {
  it('should render with default props', () => {
    const component = shallow(<NotificationsDropdown />);
    expect(component).toMatchSnapshot();
  });
});
