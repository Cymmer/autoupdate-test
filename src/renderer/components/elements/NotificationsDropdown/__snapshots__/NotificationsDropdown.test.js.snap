// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - NotificationsDropdown should render with default props 1`] = `
<div
  className="NotificationsDropdown"
>
  <IconButton
    className="NotificationsDropdown_button"
    colorName="blue"
    disabled={false}
    icon="notifications"
    iconClassName={null}
    id={null}
    isLocked={false}
    kind="button"
    onClick={[Function]}
    style={null}
    tabIndex={0}
    type="solid___md"
  />
  <Card
    className="NotificationsDropdown_dropdown"
    id={null}
    onClick={null}
    onMouseEnter={null}
    onMouseLeave={null}
  >
    <div
      className="NotificationsDropdown_head"
    >
      <Text
        className={null}
        colorClass="n___700"
        element={null}
        id={null}
        type="data___sm"
      >
        Notifications
      </Text>
    </div>
    <div
      className="NotificationsDropdown_body"
    >
      <div
        className="NotificationsDropdown_empty"
      >
        <Text
          className={null}
          colorClass="n___400"
          element={null}
          id={null}
          type="body___sm"
        >
          No Notifications
        </Text>
      </div>
    </div>
  </Card>
</div>
`;
